commands.js

```diff
@@ -286,6 +286,41 @@ var commands = exports.commands = {
 	},
 	makechatroomhelp: ["/makechatroom [roomname] - Creates a new room named [roomname]. Requires: ~"],
 
+	personalroom: function (target, room, user) {
+		// Chosen name have the same limitations as regular rooms.
+		if (target.includes(',') || target.includes('|') || target.includes('[') || target.includes('-')) {
+			return this.sendReply("Room titles can't contain any of: ,|[-");
+		}
+
+		// Check if the custom part of the room title is empty. It needs a title.
+		var id = toId(target);
+		if (!id) return this.parse('/help personalroom');
+
+		// Create the name of the room using user's ID and the chosen name.
+		var roomName = target + " (" + user.userid + "'s room)";
+		id = toId(roomName);
```

Comment:
`-` has special meaning in room IDs. No user should be able to add `-` to a room ID.